documentation/modules/configuring/con-config-mirrormaker2-high-volume-messages.adoc

```diff
@@ -6,26 +6,58 @@
 
-This type of error means that there are too many messages in the producer buffer, so they can't all be flushed before the offset flush timeout is reached.
+In certain circumstances, you might want to increase the size of the message batches sent in a single produce request.
+You increase the batch size using the `batch.size` producer configuration to reduce the number of outstanding messages ready to be sent.
+Messages being sent to the same partition are batched together.
+A larger batch size reduces the number of requests and the size of the backlog in the message queue.
+A produce request is sent to the target cluster when the batch size is reached.
+By increasing the batch size, produce requests are delayed and more messages are added to the batch and sent to brokers at the same time.  
+This can improve throughput when you have just a few topic partitions that handle large numbers of messages.  
 
-If you are getting this type of error, try the following configuration changes:
+Consider the number and size of the records that the producer handles for a suitable producer batch size. 
 
-* Decreasing the default value in bytes of the `producer.buffer.memory`
-* Increasing the default value in milliseconds of the `offset.flush.timeout.ms`
+Use `linger.ms` to add a wait time in milliseconds to delay produce requests when producer load decreases. 
+What the delay means is that more records can be added to batches if they are under the maximum batch size.  
+
+Configure the `batch.size` and `linger.ms` values at the source connector level (`producer.override.*`), as they relate to the specific producer that sends topic messages to the target Kafka cluster.
+
+The data replication pipeline to the target Kafka cluster is as follows:
+
+.Data replication pipeline
+*source topic -> (Kafka Connect tasks) source message queue -> producer buffer -> target topic*.
+
+The producer sends messages in its buffer to topics in the target Kafka cluster.
+While this is happening, Kafka Connect tasks continue to poll topics in the source Kafka cluster to add messages to the source message queue.
 
-The changes should help to keep the underlying Kafka Connect queue of outstanding messages at a manageable size.
-You might need to adjust the values to have the desired effect.
+The size of the buffer is set using the `producer.buffer.memory` property. 
+Tasks wait for a specified timeout period (`offset.flush.timeout.ms`) before the buffer is flushed. 
+This should be enough time for the sent messages to be acknowledged by the brokers and offset data committed. 
 
-If these configuration changes don't resolve the error, you can try increasing the number of tasks that run in parallel by doing the following.
+NOTE: Kafka 3.1.0 and newer versions do not wait for the producer to empty the message queue before committing offsets, except during shutdown.
```

Comment:
Yeah, I was also wondering that in a previous comment. The tuning proposed here is still useful in case of high latency, but after that release, which includes the fix, you can't be in a situation where MM2 appears as blocked (Failed to flush).

Reply:
I've removed the notes and incorporated into the body text. I've also changed the config of `linger`. 
I also removed the sentence about too many messages in the buffer so they can't all be flushed. I'm assuming the other references to flush -- in the context of handling large volumes of messages -- are okay